src/components/layout/Header/index.jsx

```diff
@@ -1,10 +1,51 @@
+'use client'
+
+import { useEffect, useState } from 'react'
 import clsx from 'clsx'
 import styles from './index.module.scss'
 
-export const Header = () => {
+export const Header = ({ headerRef }) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const toggleMenu = () => setIsOpen((prev) => !prev)
+
+  // スクロール制御
+  useEffect(() => {
+    document.body.style.overflow = isOpen ? 'hidden' : 'auto'
+    return () => {
+      document.body.style.overflow = 'auto'
+    }
+  }, [isOpen])
+
+  useEffect(() => {
+    if (!headerRef.current) return
+
+    const buttons = document.querySelectorAll('[data-scroll]')
```

Comment:
[must]
documentオブジェクトからではなく、headerRefから取得するようにしましょう！

```suggestion
    const buttons = headerRef.current.querySelectorAll('[data-scroll]')
```

documentオブジェクトはルートオブジェクト（ページ全体を指す）ので、すべての要素から`data-scroll`属性を探します。
Headerコンポーネント以外に`data-scroll`属性があった場合、意図しない要素まで取得してしまい、バグの原因になります。

headerRefはHeaderコンポーネント内の要素に限定されるので、意図しない要素が取得される心配はありません。
スコープをなるべく絞ることで、影響範囲を小さくでき、結果バグも発生しにくくなるので、他の場面でも意識してみてください！

Reply:
[3f7630c](https://github.com/gizumo-education/front-coding/pull/233/commits/3f7630c3b44249f80e4eba2472642fa2fb9bdcaa)

data-scroll属性取得をheader内に限定するように修正いたしました。